File: src/main/java/no/fint/provider/springer/model/OrganisasjonRepository.java
package no.fint.provider.springer.model;

import lombok.extern.slf4j.Slf4j;
import no.fint.event.model.Event;
import no.fint.event.model.ResponseStatus;
import no.fint.event.model.Status;
import no.fint.model.administrasjon.organisasjon.OrganisasjonActions;
import no.fint.model.resource.FintLinks;
import no.fint.model.resource.administrasjon.organisasjon.OrganisasjonselementResource;
import no.fint.provider.springer.storage.SpringerRepository;
import org.springframework.stereotype.Repository;

import java.util.Set;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Slf4j
@Repository
public class OrganisasjonRepository extends SpringerRepository {

    @Override
    public void accept(Event<FintLinks> response) {
        switch (OrganisasjonActions.valueOf(response.getAction())) {
            case GET_ALL_ORGANISASJONSELEMENT:
                query(OrganisasjonselementResource.class, response);
                break;
            default:
                response.setStatus(Status.ADAPTER_REJECTED);
                response.setResponseStatus(ResponseStatus.REJECTED);
                response.setStatusCode("INVALID_ACTION");
                response.setMessage("Invalid action");
        }
    }

    @Override
    public Set<String> actions() {
        return Stream
                .of(
                        OrganisasjonActions.GET_ALL_ORGANISASJONSELEMENT
                )
                .map(Enum::name)
                .collect(Collectors.toSet());
    }
}

File: src/main/kotlin/no/fint/provider/springer/model/OrganisasjonRepository.kt
package no.fint.provider.springer.model

import no.fint.event.model.Event
import no.fint.event.model.ResponseStatus
import no.fint.event.model.Status
import no.fint.provider.springer.storage.SpringerRepository
import no.fint.provider.springer.storage.Wrapper
import no.novari.fint.model.administrasjon.organisasjon.OrganisasjonActions
import no.novari.fint.model.resource.FintLinks
import no.novari.fint.model.resource.administrasjon.organisasjon.OrganisasjonselementResource
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.stereotype.Repository
import java.util.stream.Collectors
import java.util.stream.Stream

@Repository
class OrganisasjonRepository(
    mongoTemplate: MongoTemplate,
    wrapper: Wrapper
) : SpringerRepository(mongoTemplate, wrapper) {

    override fun accept(response: Event<FintLinks>) {
        when (OrganisasjonActions.valueOf(response.action)) {
            OrganisasjonActions.GET_ALL_ORGANISASJONSELEMENT -> query(OrganisasjonselementResource::class.java, response)
            else -> invalid(response)
        }
    }

    private fun invalid(response: Event<FintLinks>) {
        response.status = Status.ADAPTER_REJECTED
        response.responseStatus = ResponseStatus.REJECTED
        response.statusCode = "INVALID_ACTION"
        response.message = "Invalid action"
    }

    override fun actions(): Set<String> =
        Stream.of(OrganisasjonActions.GET_ALL_ORGANISASJONSELEMENT)
            .map { it.name }
            .collect(Collectors.toSet())
}
